app/client-v2/src/utils/projectStatus.ts

```diff
@@ -1,7 +1,50 @@
-import type { StatusTypes } from "@/types/projectTypes";
+import { COMPLETE_STATUS_TYPES, type AnalysisStatus, type StatusTypes } from "@/types/projectTypes";
 
 export const hasSampleFailed = (statusType: StatusTypes | undefined, cluster: string | undefined) =>
   statusType === "failed" || (statusType === "finished" && !cluster);
 
 export const hasSamplePassed = (statusType: StatusTypes | undefined, cluster: string | undefined) =>
   statusType === "finished" && !!cluster;
+
+export const hasMicroreactClusterPassed = (
+  microreactClusterStatuses: Record<string, StatusTypes> | undefined,
+  cluster: string | undefined
+): boolean => {
+  if (microreactClusterStatuses && cluster) {
+    return microreactClusterStatuses[cluster] === "finished";
+  }
+  return false;
+};
+
+export const hasMicroreactClusterFailed = (
+  status: AnalysisStatus | undefined,
+  cluster: string | undefined
+): boolean => {
+  // occurs when another sample has passed but this one fails
+  if (!cluster && isAnyMicroreactQueued(status?.microreactClusters)) return true;
+
+  if (status?.microreact == "failed" || (cluster && status?.microreactClusters?.[cluster] === "failed")) return true;
+
+  return false;
+};
+
+export const getMicroreactClusterStatus = (
+  status: AnalysisStatus | undefined,
+  cluster: string | undefined
+): StatusTypes => {
+  if (cluster && status?.microreactClusters?.[cluster]) {
+    return status.microreactClusters[cluster];
+  }
+
+  return status?.microreact || "waiting";
+};
+
+export const isAnyMicroreactQueued = (microreactClusterStatuses: Record<string, StatusTypes> | undefined): boolean => {
```

Comment:
Slightly misleading method name, since these could all be completed jobs, not currently queued.
```suggestion
export const haveAnyMicroreactBeenQueued = (microreactClusterStatuses: Record<string, StatusTypes> | undefined): boolean => {
```
Something like that?